agora_analytica/data/yle_2019.py

```diff
@@ -225,9 +235,18 @@ def process_data(df: pd.DataFrame) -> Yle2019E:
 
     # Replace with np.NaN, as Yle is using "-" to indicate skipped and no opinion
     # values
-    df = df.replace("-", np.NaN)
+    df = (df.replace("-", np.NaN))
+
+
+    #deleting the leftover candidates we can't identify (only 2 at this point)
+    l = 0
+    for i, row in df.iterrows():
+        if isinstance(row['number'], float):
+            l += 1
+            df = df.drop(i)
```

Comment:
Muutat jotain loopin sisällä, joka on big no no. Tai toisinpäin; iteroit jotain joka muuttuu. Tähän vielä liittyy teoriaa iteraattoreista, ja miten voit näennäisesti tehdä tehottoman näköistä koodia, mutta joka todellisuudessa on supertehokasta. Jos kiinnostaa niin https://www.youtube.com/watch?v=OSGv2VnC0go on pythonin iteraattoreista vastaavan lyhyt esitelmä. Pätee pitkälti myös muissakin kielissä.


Dataframesta voit suoraan filtteröidä vain rivit/sarakkeet. Se on R:stä kopioitu ominaisuus, joten ei niitä intuitiivisimpia:
```python
df = df[df['number'] != str(None)]
```
Tuo `str(None)` tulee koska combine.py:  ``df.at[i, 'number'] = str(row["election_number"]) ``